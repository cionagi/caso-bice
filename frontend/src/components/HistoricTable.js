// Dependency
import React from 'react';
import PropTypes from 'prop-types';

// Components
import HistoricTableRow from './HistoricTableRow';

const HistoricTable = ({ historicData }) => {
  const renderRows = () => {
    return historicData.map((row) => <HistoricTableRow key={row.date} historicRow={row} />);
  };

  const renderTable = () => {
    return (
      <table className="table table-striped table-hover table-sm table-dark text-center">
        <thead>
          <tr>
            <th scope="col">Fecha</th>
            <th scope="col">Valor</th>
          </tr>
        </thead>
        <tbody>{renderRows()}</tbody>
      </table>
    );
  };

  return historicData.length ? (
    renderTable()
  ) : (
    <h2 className="text-center">No se pudieron cargar los datos.</h2>
  );
};

HistoricTable.defaultProps = {
  historicData: [],
};

HistoricTable.propTypes = {
  historicData: PropTypes.array,
};

export default HistoricTable;
