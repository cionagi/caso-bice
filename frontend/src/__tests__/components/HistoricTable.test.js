import React from 'react';
import { render } from '@testing-library/react';

import HistoricTable from '../../components/HistoricTable';

test('Link renders correctly', () => {
  const { asFragment } = render(<HistoricTable />);
  expect(asFragment()).toMatchSnapshot();
});
