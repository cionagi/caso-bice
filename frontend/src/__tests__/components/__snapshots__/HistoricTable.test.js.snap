// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link renders correctly 1`] = `
<DocumentFragment>
  <table
    class="table table-striped table-hover table-sm table-dark text-center"
  >
    <thead>
      <tr>
        <th
          scope="col"
        >
          Fecha
        </th>
        <th
          scope="col"
        >
          Valor
        </th>
      </tr>
    </thead>
    <tbody />
  </table>
</DocumentFragment>
`;
